### PR TITLE
Skip update op when instance is created

### PR DESCRIPTION
**Pull request by xychu**
Also changed one test of reconcile on delete

### CODE REVIEW

File: pkg/scaffold/controller/controller.go [hunk 1]
```diff
@@ -248,20 +248,23 @@ func (r *Reconcile{{ .Resource.Kind }}) Reconcile(request reconcile.Request) (re
...
 	found := &appsv1.Deployment{}
+	created := false
 	err = r.Get(context.TODO(), types.NamespacedName{Name: deploy.Name, Namespace: deploy.Namespace}, found)
 	if err != nil && errors.IsNotFound(err) {
 		log.Info("Creating Deployment", "namespace", deploy.Namespace, "name", deploy.Name)
 		err = r.Create(context.TODO(), deploy)
 		if err != nil {
 			return reconcile.Result{}, err
+		} else {
+			created = true
```
Comment (droot) [inline pkg/scaffold/controller/controller.go]: we can probably `return reconcile.Result{}, nil` here and not change anything below ?

Comment (xychu) [reply pkg/scaffold/controller/controller.go]: @droot nice catch 👍 , I've updated.

File: pkg/scaffold/controller/controller.go [hunk 2]
```diff
@@ -251,10 +251,7 @@ func (r *Reconcile{{ .Resource.Kind }}) Reconcile(request reconcile.Request) (re
 	err = r.Get(context.TODO(), types.NamespacedName{Name: deploy.Name, Namespace: deploy.Namespace}, found)
 	if err != nil && errors.IsNotFound(err) {
 		log.Info("Creating Deployment", "namespace", deploy.Namespace, "name", deploy.Name)
-		err = r.Create(context.TODO(), deploy)
-		if err != nil {
-			return reconcile.Result{}, err
-		}
+		return reconcile.Result{}, r.Create(context.TODO(), deploy)
```
Comment (droot) [inline pkg/scaffold/controller/controller.go]: Though this is more concise, but has poor readability. I had to search where did `Create` go :)

I think the previous version is more readable.
```Go
err = r.Create(....)
return reconcile.Result{}, err
```

Comment (xychu) [reply pkg/scaffold/controller/controller.go]: Agreed and thanks.

**Pull request merged**